vmlinux/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![allow(non_upper_case_globals)]
+
+use std::env::consts::ARCH;
+use std::path::Path;
+use std::path::PathBuf;
+
+// TODO: Ideally we'd deduplicate contents at this level (as well; not just on
```

Comment:
those are also huge text files which would compress very nicely, but I assume it's impossible in const context?

Reply:
Likely not in a `const` context, yeah. We could probably have a function based interface if need be and load and cache stuff on demand. I mostly got nerd-sniped here and wanted to make it work as I had originally envisioned...

The other avenue would to exclude unnecessary headers based on features. Either way, it's probably not urgent unless/until we use this functionality in anything but example & test code.